compositor_render/src/gpu_shader/validation.rs

```diff
@@ -685,5 +703,45 @@ impl ScalarKindExt for naga::ScalarKind {
     }
 }
 
+trait ShaderParamExt {
+    fn to_string(&self) -> String;
+}
+
+impl ShaderParamExt for ShaderParam {
```

Comment:
Unless you plan on changing the `to_string`'s signature I would just use rust's `ToString` trait

Reply:
one of those types needs to be defined in this crate, neither ToString nor ShaderParam is defined here